### PR TITLE
[WIP]  Added DBAL session storaged based on PDO Session storage page

### PR DESCRIPTION
**Pull request by xocasdashdash**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes |
| Applies to | 2.1 |
| Fixed tickets | #881 |

I added a quick reference to https://github.com/symfony/symfony/blob/2.1/src/Symfony/Bridge/Doctrine/HttpFoundation/DbalSessionHandler.php basing it almost completely on the PDO Session handler

I'm merging it into this branch, even though it's not mantained anymore, because this is where this option became available to use for session storage

### CODE REVIEW

**Comment by xabbuh** [comment]
@xocasdashdash Do you think you can finish this? :)

**Comment by xocasdashdash** [comment]
@xabbuh Ok! If i don't finish it this week don't hesitate on rejecting/closing it and assigning it to somebody else!

**Comment by xocasdashdash** [comment]
@xabbuh Ok, I think i'm finished, how can I check that everything's correct format wise? How should I proceed?

File: cookbook/configuration/database_session_storage.rst [hunk 1]
```diff
@@ -1,19 +1,149 @@
...
+:class:`Symfony\\Bridge\\Doctrine\\HttpFoundation\\HttpFoundation\\DbalSessionHandler` 
+and the other uses PDO :class:`Symfony\\Component\\HttpFoundation\\Session\\Storage\\Handler\\PdoSessionHandler`.
 
+Using Doctrine to Store the Session in the Database
+---------------------------------------------------
+
+To use The DBAL session storage, you need to register a new service and configure
+ Symfony's session handling to use it:
+
+
```
Comment (WouterJ) [inline cookbook/configuration/database_session_storage.rst]: can you please remove on of these 2 empty lines?

File: cookbook/configuration/database_session_storage.rst [hunk 2]
```diff
@@ -1,19 +1,149 @@
...
+        </services>
+
+    .. code-block:: php
+
+        // app/config/config.php
+        use Symfony\Component\DependencyInjection\Definition;
+        use Symfony\Component\DependencyInjection\Reference;
+
+        $container->loadFromExtension('framework', array(
+            ...,
```
Comment (WouterJ) [inline cookbook/configuration/database_session_storage.rst]: you should use `// ...` (same on line 61)

File: cookbook/configuration/database_session_storage.rst [hunk 3]
```diff
@@ -1,19 +1,149 @@
...
+            ),
+        ));
+
+        $storageDefinition = new Definition('Symfony\Bridge\Doctrine\HttpFoundation\DbalSessionHandler', array(
+            new Reference('doctrine.dbal.default_connection'),
+        ));
+        $container->setDefinition('session.handler.dbal_handler', $storageDefinition);
+
+You can pass a second parameter to the constructor to set the table name. 
+* ``db_table``: The name of the session table in your database
```
Comment (WouterJ) [inline cookbook/configuration/database_session_storage.rst]: you should add an empty before the list. I would also prefer if this was a definition list:

````rst
``db_table``:
    The name of the session table in your database
``db_id_col``
    The name of the id column in your session table (``VARCHAR(255)`` or larger)
``db_data_col``
    The name of the value column in your session table (``TEXT`` or ``GLOB``)
``db_time_col``
    The name of the time column in your session table (``INTEGER``)
````

File: cookbook/configuration/database_session_storage.rst [hunk 4]
```diff
@@ -1,19 +1,149 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+With the given configuration, the database connection settings are the ones you've
+set for the default doctrine connection. This is OK if you're storing everything 
+in the same database. If you want to store the sessions in another database you just have
+to configure a new doctrine connection.
+
+.. note::
+
+    How to configure multiple entity managers is covered in the :doc:`/cookbook/doctrine/multiple_entity_managers` page of the book.
```
Comment (WouterJ) [inline cookbook/configuration/database_session_storage.rst]: you should add a line break after the first word that crosses the 72th character (same below)

File: cookbook/configuration/database_session_storage.rst [hunk 5]
```diff
@@ -1,19 +1,149 @@
...
+
+Table Structure and Example SQL Statements
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Because of the way this is implemented in the php class you can only configure the table name (The default is sessions)
```
Comment (WouterJ) [inline cookbook/configuration/database_session_storage.rst]: you should add an empty line before the paragraph (and wrap it, see my previous comment)

File: cookbook/configuration/database_session_storage.rst [hunk 6]
```diff
@@ -1,19 +1,149 @@
...
+Here are a couple of SQL statements to help you create a table that will work with this
+MySQL
```
Comment (WouterJ) [inline cookbook/configuration/database_session_storage.rst]: empty line missing before this headline too

File: cookbook/configuration/index.rst [hunk 7]
```diff
@@ -8,6 +8,8 @@ Configuration
     override_dir_structure
     front_controllers_and_kernel
     external_parameters
-    pdo_session_storage
+    database_session_storage
+    																																																																							
     apache_router
     web_server_configuration
+    cookbook/configuration/pdo_session_storage cookbook/configuration/database_session_storage
```
Comment (WouterJ) [inline cookbook/configuration/index.rst]: this line should be moved to the ``redirection_map`` file in the root of this repository

**Comment by WouterJ** [comment]
@xocasdashdash I've just added some formatting comments. If you fix those, you should be ready to go.

There is also a merge conflict, can you please rebase onto 2.3 (As that's the branch where this PR is going to be merged into).

Assuming `origin` points to this repository and `xocasdashdash` to your fork:
````
$ git fetch origin
$ git rebase --onto origin/2.3 origin/master
$ git push -f including_dbalsession_doc xocasdashdash
````

If you've solved the conflict, this PR can also be build by Travis, which checks for syntax errors.

Btw, why did you create a `dbal_session_storage` article with the same content as the new `database_session_storage` article?

File: cookbook/configuration/database_session_storage.rst (hunk 1 above)
Comment (xocasdashdash) [inline cookbook/configuration/database_session_storage.rst]: Ok

File: cookbook/configuration/database_session_storage.rst (hunk 2 above)
Comment (xocasdashdash) [inline cookbook/configuration/database_session_storage.rst]: which line?

File: cookbook/configuration/database_session_storage.rst (hunk 3 above)
Comment (xocasdashdash) [inline cookbook/configuration/database_session_storage.rst]: Ok

File: cookbook/configuration/database_session_storage.rst (hunk 5 above)
Comment (xocasdashdash) [inline cookbook/configuration/database_session_storage.rst]: where should the empty line go?

File: cookbook/configuration/database_session_storage.rst (hunk 6 above)
Comment (xocasdashdash) [inline cookbook/configuration/database_session_storage.rst]: Ok

File: cookbook/configuration/index.rst (hunk 7 above)
Comment (xocasdashdash) [inline cookbook/configuration/index.rst]: Ok

**Comment by xocasdashdash** [comment]
@WouterJ I've tried to fix the merge conflicts, but I can't seem to get it right :confused:

File: cookbook/configuration/database_session_storage.rst [hunk 8]
```diff
@@ -1,19 +1,153 @@
 .. index::
     single: Session; Database Storage
 
-How to Use PdoSessionHandler to Store Sessions in the Database
-==============================================================
+How to Store Session in the Database
```
Comment (timglabisch) [inline cookbook/configuration/database_session_storage.rst]: Session`s`

**Comment by xabbuh** [comment]
@xocasdashdash To be build on Travis, your pull request has to be rebased onto the branch you want to merge in (`master` in your case). Alternatively, you can open a new pull request with your (already rebased on the `2.3` branch) changes for the `2.3` branch.

**Comment by WouterJ** [comment]
Hi @xocasdashdash!

Do you have time to update your PR? We have a [doc sprint day](http://symfony.com/blog/symfony-docs-hack-day-needs-you-on-may-23rd) on May 23rd and we would like to finish as much PRs as possible before/during that PR.

Don't worry if you have no time soon, we know it's all freetime work. @javiereguiluz already started a new PR without knowing this PR existed: #5019 you can also comment on that PR with some details that you think are missing from his PR and close this PR.

If you need any help, you can almost always find one of us on the #symfony-docs IRC freenode channel.

Thanks!

**Comment by xocasdashdash** [comment]
@WouterJ I've checked @javiereguiluz PR and it looks fine by me. 

I'm gonna close down this PR then!

**Pull request closed without merging**